### PR TITLE
Snap charm

### PR DESCRIPTION
**Pull request by kngu9**

### CODE REVIEW

File: charm/sisyphus/actions/start [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+]
+if all((os.path.isfile(SISYPHUS_CERT),
+       os.path.isfile(SISYPHUS_KEY),
+       os.path.isfile(SISYPHUS_CA_CERT))):
+    cmd += [
+        '--kafka-cert', SISYPHUS_CERT,
+        '--kafka-key', SISYPHUS_KEY,
+        '--kafka-ca', SISYPHUS_CA_CERT
+    ]
+cmd += ['&']
```
Comment (cmars) [inline charm/sisyphus/actions/start:67]: I don't think this is necessary.. Popen doesn't block on the process terminating. https://docs.python.org/3/library/subprocess.html?highlight=subprocess#subprocess.Popen

File: charm/sisyphus/actions/stop [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import sisyphus_utils
+
+from charms.layer.sisyphus import Sisyphus
+
+
+sisyphus = Sisyphus()
+if not sisyphus.is_running():
+    sisyphus_utils.fail('sisyphus is not running on this unit')
+
+os.kill(sisyphus.get_pid(), signal.SIGTERM)
```
Comment (cmars) [inline charm/sisyphus/actions/stop]: Needs to deal with the case where the pid in the lock file is no longer running:
```
>>> os.kill(99999999, 15)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ProcessLookupError: [Errno 3] No such process
```

File: charm/sisyphus/actions.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+start:
+  description: Starts sisyphus
+  params:
```
Comment (cmars) [inline charm/sisyphus/actions.yaml]: I think these could be config options. I don't see much need to change these very often, and the action would be easier to run with no params. Thoughts?

File: charm/sisyphus/layer.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+- 'layer:basic'
+- 'interface:tls-certificates'
+- 'layer:tls-client'
+- 'layer:snap'
+- 'interface:kafka'
+
+options:
+  basic:
+    packages:
+      - default-jre-headless
```
Comment (cmars) [inline charm/sisyphus/layer.yaml]: sisyphus does not need java

File: charm/sisyphus/reactive/sisyphus.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    set_flag('sisyphus.configured')
+    hookenv.status_set('waiting', 'waiting for further commands')
+
+
+@when_not('sisyphus.installed')
+def install_sisyphus():
+    install()
+    set_flag('sisyphus.installed')
+
+
```
Comment (cmars) [inline charm/sisyphus/reactive/sisyphus.py:28]: Need to handle upgrade-charm.

**Review by cmars — CHANGES_REQUESTED** [review]

File: charm/sisyphus/actions/start (hunk 1 above)
Comment (kngu9) [reply charm/sisyphus/actions/start:67]: I tried, and even though Popen is non blocking, the process exits immediately if you don't specify to place the process in the background.

File: charm/sisyphus/actions/stop (hunk 2 above)
Comment (kngu9) [reply charm/sisyphus/actions/stop]: @cmars It does a check on line 12 to check whether it is running

**Review by alesstimec — APPROVED** [review]

**Review by cmars — APPROVED** [review]

**Pull request merged**